parsl/executors/high_throughput/interchange.py

```diff
@@ -68,6 +69,7 @@ class Interchange:
     4. Service single and batch requests from workers
     """
     def __init__(self,
+                 cert_dir: Optional[str],
                  client_address: str = "127.0.0.1",
```

Comment:
I did this to limit the chances of mistakenly setting `cert_dir` to `None`, which will disable encryption. Also, Ben has expressed a desire to remove all of the defaults in the interchange's initializer. I don't believe the interchange is used outside of the HTEX, so I don't have strong feelings either way here.

Reply:
Hmm; then I still suggest to put the `cert_dir` at the bottom, but I don't know if we're ready to remove the default just yet.  That is, if we assume that no one outside of Parsl internals uses `Interchange`, then this is moot point.  On the other hand, if we can't assume that, then this should go at the end of the list (and thus requires a default).

What about using the same default as elsewhere:
```python
def __init__(
    self,
    client_address: str = "127.0.0.1",
    ...,
    cert_dir: Optional[str] = "certificates"
```
This would be in keeping with the relative path to which `Executor().run_dir` defaults:
```python
    self.run_dir = '.'
```

Reply:
For context: I have a strong desire (distinct from this PR) to remove defaults and move this to kwargs if possible. c.f. PR #2973 which removes defaults from the manager - defaults that "should never be used" and that fired only in the case of a coding error that broke a feature (rather than giving an error about unspecified parameters).

In general, no one should be launching an Interchange except except exactly the corresponding version of the HighThroughputExecutor class: there's no external API presented here.

 That isn't related to this PR #3030 though.

Reply:
>> That is, if we assume that no one outside of Parsl internals uses Interchange, then this is moot point.
>
> ... there's no external API presented here.

If this is the case and or expectation, then my suggestion is moot.

Reply:
for the avoidance of doubt, I would prefer default fiddling to happen away from this PR in a specialised PR that replicates #2973

Reply:
I like the `certificates` idea, so I'll go with that